### ISSUE TITLE
@ByVal CGAffineTransform isn't passed correctly on ARM64

### ISSUE THREAD

**Issue by ntherning** [robovm/robovm#742]
Here's my test code:

```java
    public static void main(String args[]) throws Exception {
        CGPoint p = new CGPoint(10, 20);
        System.out.println(p.toString());

        CGRect r = new CGRect(10, 20, 30, 40);
        System.out.println(r.toString());

        CGAffineTransform t = new CGAffineTransform(0, -1.0, 1.0, 0, 0, 0);
        System.out.println(t.toString());
    }
```

The `CGPoint` and `CGRect` objects can be converted to strings without problem. The `CGAffineTransform.toString()` method crashes on ARM64. Looking at what clang produces for C functions which take `CGAffineTransform` by value on ARM64 it is large enough to be copied to memory and passed as a pointer. We don't do that currently. Using `byval` in the bitcode we produce doesn't help. Looks like we need to do the copy ourselves on ARM64 and pass the pointer.

**Comment by kirillp** [comment]
You might be looking at C calling conventions. Returning a Struct from a function is equivalent to passing the pointer to result in 1st invisible parameter (like this). 
so that RoboVM can 
  - allocate the struct data on GC heap
  - pass it directly to the calling function to obtain the result without having any copy

**Comment by ntherning** [comment]
Yeah, it's all about implementing the ABI properly for each platform. E.g. small structs can on some platforms be returned in registers. Seems like we don't handle large structs on ARM64 properly. The [ARM64 (Aarch64) ABI](http://infocenter.arm.com/help/topic/com.arm.doc.ihi0055b/IHI0055B_aapcs64.pdf) specifies that structs larger than 16 bytes should be passed in memory and returned in an invisible 1st parameter. We need to get LLVM to generate the proper machine code for us.

**Comment by ntherning** [comment]
Fixed now with #754.

**Comment by kirillp** [comment]
I made a brief unprofessional view on the change and have a question:

as far as I get you create a local variable on the stack via alloca and then invoke the method and copy the result to a heap object. It is possible to invoke the method directly with the pointer to heap object avoiding the unnecessary copy ?
